File: 01/01.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"regexp"
	"strconv"
	"strings"
)

func getSum(data string) int {
	lines := strings.Split(data, "\n")

	sum := 0
	for _, line := range lines {
		re := regexp.MustCompile("[0-9]")
		numbers := re.FindAllString(line, -1)

		if len(numbers) > 0 {
			firstNum, _ := strconv.Atoi(numbers[0])
			lastNum, _ := strconv.Atoi(numbers[len(numbers)-1])

			lineNumber := firstNum*10 + lastNum
			sum += lineNumber
		}
	}
	return sum
}

func main() {
	file, err := os.Open("./input1.txt")
	if err != nil {
		fmt.Println("Unable to open file!")
		os.Exit(1)
	}
	defer file.Close()

	scanner := bufio.NewScanner(file)
	data := ""
	for scanner.Scan() {
		data += scanner.Text() + "\n"
	}

	output := getSum(data)
	fmt.Println("final sum: ", output)
}

File: 01/01.php
<?php

function getSum($data) {
    $separator = "\r\n";
    $line = strtok($data, $separator);

    $sum = 0;
    while ($line !== false) {
        preg_match_all('/[0-9]/', $line, $numbers); # get each line
        $numbers = array_filter($numbers); # get numbers in array

        $firstNum = reset($numbers[0]); # get first number
        $lastNum = end($numbers[0]); # get last number

        $lineNumber = $firstNum * 10 + $lastNum; # combine numbers        
        $sum = $sum + $lineNumber;
        # get next line
        $line = strtok( $separator );
    }
    echo "final sum: " . $sum;
}
function test() {
    $input = "1abc2\npqr3stu8vwx\na1b2c3d4e5f\ntreb7uchet\nasefe1\nasef11pogh";

    getSum($input);
}

// test();
$myfile = fopen("input1.txt", "r") or die("Unable to open file!");
getSum(fread($myfile,filesize("input1.txt")));
fclose($myfile);

# 12142132 is too high
?>